packages/cli/src/commands/generate/auth/templates/auth.firebase.js.template

```diff
@@ -0,0 +1,45 @@
+// Define what you want `currentUser` to return throughout your app. For example,
+// to return a real user from your database, you could do something like:
+//
+//   export const getCurrentUser = async ({ email }) => {
+//     return await db.user.findOne({ where: { email } })
+//   }
+
+import { AuthenticationError } from '@redwoodjs/api'
+import admin from 'firebase-admin'
+
+import { db } from './db'
+
+const config = {
+  apiKey: process.env.FIREBASE_API_KEY,
+  authDomain: process.env.FIREBASE_AUTH_DOMAIN,
+  databaseURL: process.env.FIREBASE_DATABASE_URL,
+  projectId: process.env.FIREBASE_PROJECT_ID,
+  storageBucket: process.env.FIREBASE_STORAGE_BUCKET,
+  messagingSenderId: process.env.FIREBASE_MESSAGING_SENDER_ID,
+  appId: process.env.FIREBASE_APP_ID,
+}
+
+const adminApp = admin.initializeApp(config)
+
+export const getCurrentUser = (token) =>
+  adminApp
+    .auth()
+    .verifyIdToken(token)
+    .then(({ email, uid }) =>
+      db.user
```

Comment:
We can't use this code in the actual template—there's no guarantee the app will have a `user` model, or any concept of a user at all, really. I think we need to leave the implementation as example code in the comment at the top. `getCurrentUser` should just return the minimum needed to get *something* useful out of the token. Like maybe `verifyIdToken` and then just return `email` and `uid` as a simple object?

Reply:
Makes sense, proposition for getCurrentUser:
`
export const getCurrentUser = (token) =>
  adminApp
    .auth()
    .verifyIdToken(token)
    .then(({ email, uid }) => ({email, uid})
    )
`
if it's ok I'll update the branch.

Reply:
It's ok to use async await here too, since it makes the code slightly flatter and easier to reason about:

```js
export const getCurrentUser = async (token) => {
  const { email, uid } = await adminApp.auth().verifyToken(token)
  return { email, uid }
}
```

Reply:
Agreed @peterp , I pushed this for a fix, much easier indeed :)